### DISCUSSION TITLE
Allow to specify a bool option as a key value pair

### DISCUSSION THREAD

**Discussion by lorenzwalthert** [fastapi/typer#704] [Questions]
### First Check

- [X] I added a very descriptive title here.
- [X] I used the GitHub search to find a similar question and didn't find it.
- [X] I searched the Typer documentation, with the integrated search.
- [X] I already searched in Google "How to X in Typer" and didn't find any information.
- [X] I already read and followed all the tutorial in the docs and didn't find an answer.
- [X] I already checked if it is not related to Typer but to [Click](https://github.com/pallets/click). Not sure it is but since typer does some type conversion given the type annotation, I think it could handle that complexity.

### Commit to Help

- [ ] I commit to help with one of those options 👆
**I could not find an easy issue where I could contribute. But I did a one time sponsorship of $ 20.- to make up for that.**

### Example Code

```python
import typer 

app = typer.Typer()


@app.command()
def hello(is_true: bool = typer.Option(True)):
    return 


if __name__ == "__main__":
    app()
```

Then, calling the script:
```bash
$ python scratch/typer-key-value.py --is-true=False
Error: Option '--is-true' does not take a value
```

Other ideas such as changing the type annotation to `str` is also hacky.

### Description

I would like to specify a bool option as a key value pair. The use case for this is that in certains circumstances, it's only possible to specify key value pairs for a command line program. In my case, starting a training job on Sagemaker, the hyper-parameters have to be supplied as key-value pairs. So if I have a boolean hyper-parameter, I must type it as a string and then later convert it, which is a workaround.

### Operating System

macOS

### Typer Version

0.9.0

### Python Version

3.11.4

**Comment by lee-cq** [comment]
I tested the following code:
```python
@app.command(name="test")
def tes(ver: bool = Option(
    False, "--verify", "-v",
    help="Verify SSL Certificates"
)):
    print("ver is", ver)
```
run  `__main__.py test -v `, it works `ver is True`.

and  `typer.Option()` has a parameter `is_flag`, It will be a flag.
like this: 
```python
@app.command(name="test")
def tes(ver: bool = Option(
    False, "--verify", "-v",
    is_flag=True,
    help="Verify SSL Certificates"
)):
    print("ver is", ver)

**Comment by lorenzwalthert** [comment]
But what I want to run is `__main__.py test --verify=False`. So you mean probably `is_flag=False`? If you edit your answer, I'll accept it 😄 .

```python
@app.command(name="test")
def tes(ver: bool = Option(
    False, "--verify", "-v",
    is_flag=False,
    help="Verify SSL Certificates"
)):
    print("ver is", ver)
```

**Comment by riklopfer** [comment]
Poking around in the code, I see that `is_flag` is not supported, but I don't get the warning. 

```
            warnings.warn(
                "The 'is_flag' and 'flag_value' parameters are not supported by Typer "
                "and will be removed entirely in a future release.",
                DeprecationWarning,
                stacklevel=2,
            )

```

Does this mean that passing booleans as values is not supported?

**Reply by lorenzwalthert** [reply]
Seems like support for it was removed in https://github.com/fastapi/typer/pull/987, but there is an ongoing discussion about the problems that arise with removal.
